cloudify_cli/execution_events_fetcher.py

```diff
@@ -101,7 +131,10 @@ def wait_for_execution(client,
     events_fetcher = ExecutionEventsFetcher(client, execution.id,
                                             include_logs=include_logs)
 
-    # Poll for execution status until execution ends
+    # Poll for execution status and execution logs, until execution ends
+    # and we receive an event of type in WORKFLOW_END_TYPES
+    execution_ended = False
+    events_watcher = EventsWatcher(events_handler)
     while True:
```

Comment:
hmm, not sure how to write that, while still keeping the original behaviour, and making it more readable

all changes would change the behaviour of the function slightly, maybe only in unrealistic hypothetical cases, but still...

a few possible variants in pseudocode here https://gist.github.com/tehasdf/c65a833321ed4e15566eec75e871ce89
i could rewrite it as the while_condition_sleep_first version version but i'm not sure if it's really worth it